src/NServiceBus.TransportTests/When_multiple_messages_are_available_and_concurrency_is_increased_and_decreased_after_pump_started.cs

```diff
@@ -0,0 +1,66 @@
+﻿namespace NServiceBus.TransportTests;
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using Transport;
+
+public class When_multiple_messages_are_available_and_concurrency_is_increased_and_decreased_after_pump_started : NServiceBusTransportTest
+{
+    [TestCase(TransportTransactionMode.None)]
+    [TestCase(TransportTransactionMode.ReceiveOnly)]
+    [TestCase(TransportTransactionMode.SendsAtomicWithReceive)]
+    [TestCase(TransportTransactionMode.TransactionScope)]
+    public async Task The_last_concurrency_value_should_be_respected(TransportTransactionMode transactionMode)
+    {
+        const int startConcurrencyLevel = 5;
+        const int newConcurrencyLevel = 10;
+        const int lastConcurrencyLevel = 1;
+        var onMessageCalls = new ConcurrentQueue<TaskCompletionSource>();
+
+        await StartPump(async (context, _) =>
+        {
+            var tcs = CreateTaskCompletionSource();
+            onMessageCalls.Enqueue(tcs);
+            // "block" current pipeline invocation
+            await tcs.Task;
+        },
+            (errorContext, __) => throw new Exception("unexpected error", errorContext.Exception),
+            transactionMode,
+            pushRuntimeSettings: new PushRuntimeSettings(startConcurrencyLevel));
+
+        await receiver.ChangeConcurrency(new PushRuntimeSettings(newConcurrencyLevel));
+        await receiver.ChangeConcurrency(new PushRuntimeSettings(lastConcurrencyLevel));
+
+        for (int i = 0; i < startConcurrencyLevel * 2; i++)
+        {
+            await SendMessage(InputQueueName);
+        }
+
+        // we need to wait because it might take a bit till the pump has invoked all pipelines
+        while (onMessageCalls.IsEmpty)
```

Comment:
is not clearer to perform the check based on the last concurrency level?

`while (onMessageCalls.Count < lastConcurrencyLevel)`

Reply:
The IDE is smart enough to [know that](https://github.com/Particular/NServiceBus/pull/7124/files/91f2235bae228b4bb1d3ced9aa47be1c6c3ad30e#diff-a36ec1c18d96acfc3f10595e65601c5c1f4a61b191e9d390068edd4a140e237fR20) `lastConcurrencyLevel` is set to `1` so it complains about this format.

I'll change `lastConcurrencyLevel` to be 2 rather, then we can use the more readable syntax